### DISCUSSION TITLE
Frequently Asked Questions

### DISCUSSION THREAD

**Discussion by cocoahero** [Shopify/storefront-api-feedback#5] [Help] [answered]
Welcome to the our Github Discussions for the Storefront API

Please read through this information prior to asking your question.

**What is a custom storefront?**
- Think mobile apps for your business, a custom front end for your website, adding commerce into a video game, offline commerce such as a smart mirror - and more.
- If you have questions about customizing your theme please post in the[ Shopify Design board](https://community.shopify.com/c/shopify-community/ct-p/en)

**How should I report a bug?**
If you are reporting a bug with a specific SDK please utilize the Github repos. Otherwise, please post it under the "Help" category with the title [bug] followed by a brief description.

**What are the rate limits?**
* The Storefront API scales to support your largest sale and does not have rate limits 
* Bot protection runs on the Storefront API to protect your app. 

**How do I handle authentication?**
The Storefront API has the following access options: 
* Public access: Used to query the API from a browser or mobile app.
* Private access: Used to query the API from a server or other private context, like a Hydrogen backend.

You can create an access token in the [Headless Channel](https://shopify.dev/docs/custom-storefronts/building-with-the-storefront-api/manage-headless-channels). For a public app or Sales Channel, the storefront access token is obtained through OAuth

**How do I manage a Cart and Checkout?**
* The Storefront API returns a Web URL field that your customer will complete the checkout process on, you can get this web URL and redirect a shopper from the cart to the web checkout once they are ready to enter the checkout flow.

**Can I do a flash sale using the Storefront API?**
Yes!

**Comment by morganmccunn** [comment, marked as answer]
We'll update this over time!
